src/paths/shortest_paths.c

```diff
@@ -1275,3 +1276,326 @@ int igraph_diameter_dijkstra(const igraph_t *graph,
+        VECTOR(*weights)[eid] = IGRAPH_INFINITY;
+    }
+    igraph_vector_destroy(&eids);
+    IGRAPH_FINALLY_CLEAN(1);
+    return IGRAPH_SUCCESS;
+}
+
+/**
+ * \function igraph_k_shortest_paths
+ * \brief K shortest paths (geodesics) between two vertices.
+ *
+ * \param graph The graph object.
+ * \param weights The edge weights of the graph. Can be \c NULL for an
+ *        unweighted graph. Infinite weights will be treated as missing
+ *        edges.
+ * \param paths Pointer to an initialized pointer vector, the result
+ *        will be stored here in \ref igraph_vector_t objects. Each vector
+ *        object contains the edges along the k-th shortest path between the
+ *         \p from and \p to vertex, where k is the pointer vector index.
+ * \param k The number of paths. 
+ * \param from The id of the vertex from which the geodesics are
+ *        calculated.
+ * \param to The id of the vertex to which the geodesics are
+ *        calculated.
+ * \param mode The type of shortest paths to be used for the
+ *        calculation in directed graphs. Possible values:
+ *        \clist
+ *        \cli IGRAPH_OUT
+ *          The shortest outgoing paths are calculated.
+ *        \cli IGRAPH_IN
+ *          The shortest incoming paths are calculated.
+ *        \cli IGRAPH_ALL
+ *          The directed graph is considered as an
+ *          undirected one for the computation.
+ *        \endclist
+ * \return Error code:
+ *        \clist
+ *        \cli IGRAPH_ENOMEM
+ *           Not enough memory for temporary data.
+ *        \cli IGRAPH_EINVVID
+ *           \p from or \p to is an invalid vertex id.
+ *        \cli IGRAPH_EINVMODE
+ *           Invalid mode argument.
+ *        \cli IGRAPH_EINVAL
+ *           Invalid argument.
+ *        \endclist
+ *
+ * Time complexity: K|V|(|V|log|V| + |E|), where |V| is the number of vertices,
+ *                  and |E| is the number of edges.
+ */
+int igraph_k_shortest_paths(const igraph_t *graph,
+                            igraph_vector_t *weights,
+                            igraph_vector_ptr_t *paths,
+                            igraph_integer_t k,
+                            igraph_integer_t from,
+                            igraph_integer_t to,
+                            igraph_neimode_t mode) {
+    igraph_vector_ptr_t paths_pot; /* potential shortest paths */
+    igraph_integer_t vertex_spur;
+    igraph_vector_t path_spur;
+    igraph_vector_t *path_root, *path_total;
+    igraph_integer_t nr_edges_root, i_path_current, i_path, edge_path_root, vertex_root_del;
+    long int i;
+    igraph_vector_t weights_old;
+    igraph_vector_t edges_removed;
+    igraph_bool_t added_path, null_weights;
+    long int nr_edges = igraph_ecount(graph);
+    igraph_bool_t infinite_path, already_in_potential_paths;
+    igraph_vector_t *path_0;
+
+    igraph_vector_ptr_free_all(paths);
+    IGRAPH_VECTOR_PTR_SET_ITEM_DESTRUCTOR(paths, igraph_vector_destroy);
```

Comment:
Do not change the item destructor permanently.

Reply:
Not relevant any more as we've switched to `igraph_vector_int_list_t`.